### ISSUE TITLE
A New Sample Attribute type that means and validates against a min/max range

### ISSUE THREAD

**Issue by stuzart** [seek4science/seek#791]
rather than just saying it is an Integer, also specify the range. The range would be set when creating the attribute

**Comment by alaninmcr** [comment]
Do not limit it to integers, a real number could also have this. Maybe data, date/time :-)

**Comment by rabuono** [comment]
Why not expand this to be a User input rule?
A User inputs a rule written as a regex for the Attribute.
A regex Attribute Type could accommodate a large variety of restrictions Users might come up with, without the need to create multiple new Attribute Types.

**Comment by fbacall** [comment]
I don't think a regex would work to validate a range, but I agree that it should be an optional restriction on the existing numeric types intead of a new attribute type

**Comment by rabuono** [comment]
What I mean is that a regex would allow Users to cover a few cases more, and wouldn't lead to many new attribute types.

For example:
For positive integers from 0-399 and "NA" for not available: `(?<!-)\b([1-3]?\d{1,2}|NA)\b`
Thus not having to create a attribute type for a range plus a specific value

The downside is that I don't know if SEEK would understand/categorize the attribute as an `integer` attribute. But, following the idea an added restriction to an existing type, a regex would be used to further restrict a `number` attribute.

**Comment by alaninmcr** [comment]
I don't think it is reasonable to expect a user to enter that regular expression. So you would need several adapters that transformed a range into a regular expression.

**Comment by fbacall** [comment]
That works at a technical level for validation, but I think there are very few users who would be able to produce a regex like that (what Alan said).

Also if we just stored the regex, we would not be able to generate nice UI hints, like having text that says "Input a value between 0 and 399", or have validation in the UI that will only let you input values within the range, or do other things like produce a JSON schema and display meaningful error messages when validation fails ("Input out of range" vs. "Input does not match pattern")
